python_modules/libraries/dagster-airflow/dagster_airflow/airflow_dag_converter.py

```diff
@@ -214,162 +162,19 @@ def _solid(context):  # pylint: disable=unused-argument
-    # consistency with how execution_date is set for manually triggered tasks, i.e.
-    # triggered_date == execution_date.
-    if dag_run and dag_run.external_trigger:
-        prev_execution_date = execution_date
-        next_execution_date = execution_date
-    else:
-        prev_execution_date = task.dag.previous_schedule(execution_date)
-        next_execution_date = task.dag.following_schedule(execution_date)
-
-    next_ds = None
-    next_ds_nodash = None
-    if next_execution_date:
-        next_ds = next_execution_date.strftime("%Y-%m-%d")
-        next_ds_nodash = next_ds.replace("-", "")
-        next_execution_date = pendulum.instance(next_execution_date)
-
-    prev_ds = None
-    prev_ds_nodash = None
-    if prev_execution_date:
-        prev_ds = prev_execution_date.strftime("%Y-%m-%d")
-        prev_ds_nodash = prev_ds.replace("-", "")
-        prev_execution_date = pendulum.instance(prev_execution_date)
-
-    ds_nodash = ds.replace("-", "")
-    ts_nodash = execution_date.strftime("%Y%m%dT%H%M%S")
-    ts_nodash_with_tz = ts.replace("-", "").replace(":", "")
-    yesterday_ds_nodash = yesterday_ds.replace("-", "")
-    tomorrow_ds_nodash = tomorrow_ds.replace("-", "")
-
-    ti_key_str = "{dag_id}__{task_id}__{ds_nodash}".format(
-        dag_id=task.dag_id, task_id=task.task_id, ds_nodash=ds_nodash
-    )
-
-    if task.params:
-        params.update(task.params)
-
-    return {
-        "dag": task.dag,
-        "ds": ds,
-        "next_ds": next_ds,
-        "next_ds_nodash": next_ds_nodash,
-        "prev_ds": prev_ds,
-        "prev_ds_nodash": prev_ds_nodash,
-        "ds_nodash": ds_nodash,
-        "ts": ts,
-        "ts_nodash": ts_nodash,
-        "ts_nodash_with_tz": ts_nodash_with_tz,
-        "yesterday_ds": yesterday_ds,
-        "yesterday_ds_nodash": yesterday_ds_nodash,
-        "tomorrow_ds": tomorrow_ds,
-        "tomorrow_ds_nodash": tomorrow_ds_nodash,
-        "END_DATE": ds,
-        "end_date": ds,
-        "dag_run": dag_run,
-        "run_id": run_id,
-        "execution_date": pendulum.instance(execution_date),
-        "prev_execution_date": prev_execution_date,
-        "prev_execution_date_success": lazy_object_proxy.Proxy(
-            lambda: task_instance.previous_execution_date_success
-        ),
-        "prev_start_date_success": lazy_object_proxy.Proxy(
-            lambda: task_instance.previous_start_date_success
-        ),
-        "next_execution_date": next_execution_date,
-        "latest_date": ds,
-        "macros": macros,
-        "params": params,
-        "tables": tables,
-        "task": task,
-        "task_instance": task_instance,
-        "ti": task_instance,
-        "task_instance_key_str": ti_key_str,
-        "test_mode": task_instance.test_mode,
-        "inlets": task.inlets,
-        "outlets": task.outlets,
-    }
+        with replace_airflow_logger_handlers():
+            if str(airflow_version) >= "2.0.0":
```

Comment:
This looks like it is preexisting, but doing semver comparison with strcomp seems problematic. Would be nice to fix in follow up